docs/source/models.rst

```diff
@@ -309,47 +238,54 @@ saved model, use the :py:func:`set_signature() <mlflow.models.set_signature>` AP
-field should be ``None`` for scalar parameters, or ``(-1,)`` for list parameters.
-See some examples of constructing them below.
 
-Column-based Signature Example
+A model signature consists of three schemas **(1) inputs**, **(2)outputs**, and **(3) inference params**. The
+former two simply refer to the input and output of the model, while the inference params is a set of additional params 
+that are passed to the model at inference time. 
+
+Input and output schemas can be either column-based or tensor-based. 
+
+Column-based Signature
 """"""""""""""""""""""""""""""
-Each column-based input and output is represented by a type corresponding to one of
-:py:class:`MLflow data types <mlflow.types.DataType>` and an optional name. Input columns can also be marked
-as ``optional``, indicating whether they are required as input to the model or can be omitted. The following example
-displays a modified MLmodel file excerpt containing the model signature for a classification model trained on
-the `Iris dataset <https://archive.ics.uci.edu/ml/datasets/iris>`_. The input has 4 named, numeric columns and 1 named,
-optional string column.
-The output is an unnamed integer specifying the predicted class.
+
+This signature is typically used for traditional machine learning models that take tabular data as input such as Pandas 
+DataFrame. Column-based schemas are a sequence of (optionally) named columns with data type. Each column-based input and 
+output is represented by a type corresponding to one of :ref:`supported data types <supported-data-types>` and an 
+optional name. Input columns can also be marked as ``optional``, indicating whether they are required as input to the 
+model or can be omitted. 
+
+The following example displays the model signature for a classification model trained on the 
+`Iris dataset <https://archive.ics.uci.edu/ml/datasets/iris>`_. The input has 4 named, numeric columns and 
+1 named, optional string column. The output is an unnamed integer specifying the predicted class.
 
 .. code-block:: yaml
 
   signature:
-      inputs: '[{"name": "sepal length (cm)", "type": "double"}, {"name": "sepal width
-        (cm)", "type": "double"}, {"name": "petal length (cm)", "type": "double"}, {"name":
-        "petal width (cm)", "type": "double"}, {"name": "class", "type": "string", "optional": "true"}]'
+      inputs: '[
+        {"name": "sepal length (cm)", "type": "double"}, 
+        {"name": "sepal width(cm)",   "type": "double"}, 
+        {"name": "petal length (cm)", "type": "double"}, 
+        {"name": "petal width (cm)",  "type": "double"}, 
+        {"name": "class",             "type": "string",  "optional": "true"}
+      ]'
       outputs: '[{"type": "integer"}]'
       params: null
 
-Tensor-based Signature Example
+Tensor-based Signature
 """"""""""""""""""""""""""""""
+This signature is typically used for deep learning models that take tensors as input such as images, speech, etc. 
```

Comment:
It's not just for deep learning models, could we just use `models`?

Reply:
That's right, probably sth like `models that take tensors such as imagesa and speech as input, as represented by deep learning`.